certificate-renewal/REPWRITERSPECS/BANNO.CDRENEW.V1.POW

```diff
@@ -944,15 +952,27 @@ PROCEDURE READCONFIGFILESETTINGS
          USEROPTIONSCHR(6)=PARAMVALUE
         END
 
-[* Debug mode override
+[* Program Installation Date
+*]
+       ELSE IF PARAMNAME="PID" AND
+               PARAMVALUE<>"" THEN
+        PARAMINSTALLDATE=DATEVALUE(PARAMVALUE)
+
+[* Debug mode. This is a hidden parameter setting which can override the default debug
+** mode setting. Use this "DB:" parameter setting in the CFG Letter file with "ON"
+** or "OFF" to force the debug mode to be other than the default value of being on for
+** the first 90 days after the most recent program modification date.
 *]
-       ELSE IF PARAMNAME="DB" THEN
+       ELSE IF PARAMNAME="DB" AND
+               PARAMVALUE<>"" THEN
         DO
-         IF PARAMVALUE="TRUE" THEN
-          BNODEBUGMODE=TRUE
-         ELSE IF PARAMVALUE="FALSE" THEN
-          BNODEBUGMODE=FALSE
-       END
+         PARAMDEBUGMODEFOUND=TRUE
+         IF CHARACTERSEARCH("ONTRUEYES",UPPERCASE(PARAMVALUE))>0 THEN
+          PARAMDEBUGMODE=BNODEBUGMODEON
+         ELSE IF CHARACTERSEARCH("OFFFALSENO",UPPERCASE(PARAMVALUE))>0 THEN
```

Comment:
Can we just get rid of this and assume it is off if it is not one of the valid ON values?  And then just make sure you are initializing PARAMDEBUGMODE.

Reply:
@rrobison44 leaving this in actually also allows the 'hidden' DB parameter setting - if used - to force debug mode off. It just seemed logical that the DB parameter switch should also allow the CU to turn the debug mode off, overriding the standard 90 days default of it being on or overriding the Program Installation Date, forcing the debug mode on as well.  It also allows the developers or BA to easily turn the debug code off thus allowing them to just see the pertinent JSON code when reviewing it for info. Let me know if you still want it out.

Reply:
I think it works fine as is.  But since you already have PARAMDEBUGMODEFOUND, and you are setting it to TRUE if DB is found and it has a value, then what you are describing still works without having the ELSE as long as you are initializing PARAMDEBUGMODE to OFF.  If they have some rogue value in DB (like "disabled") then I think the code still functions as if we are overriding debug mode with the DB param config and setting it to OFF -- as that would be the default.  i.e. having DB set to some non-blank value (other than ON/TRUE/YES) effectively forces debug mode to OFF.  It doesn't necessarily have to be OFF/FALSE/NO.  But I am good with whatever you decide.  I just think you can get rid of some code and still do, logically, the exact same thing you are doing with the ELSE clause.  Of course it is also possible that my thinking is off :)

Reply:
@rrobison44 
**I'll certainly take another look at what I have.** 

My goal was to have the following functionality:
1. Debug mode defaults to being off
2. Determine if debug mode should be turned on based upon the last mod date and the program installation date values. Turn on if warranted.
3. Check for a 'DB:' value in the CFG Letter file
- If there is no DB value in the CFG Letter file, or a DB parameter setting with an invalid value (or no value at all), use the last mod date/installation date logic results.
- If there is a DB value with a valid YES/ON/TRUE value then override the current (last mod date/installation date) logic and force the debug mode on
- If there is a DB value with a valid NO/OFF/FALSE value then override the current logic and force the debug mode off.

I was trying to avoid the "unless I specifically turn it on, I mean for it to be off" logic.

I'll take a further look at it.

Reply:
@rrobison44 I know you already approved this but taking your suggestion, I was able to 'condense' the logic a bit further and remove a few lines of the code. The logic is still behaving as expected. Thanks for the nudge!